### PR TITLE
CONCF-796 Unit tests for Curated Main Pages

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/CONCF-796

Added 24 unit tests related to Curated Main Pages and fixed some minor issues that I bumped into.

/cc @Warkot @bognix @vforge

### CODE REVIEW

**Comment by tomasznapieralski** [comment]
LGTM

File: test/specs/front/scripts/main/components/TrendingVideosItemComponent.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+	equal(componentMock.get('thumbUrl'), 'http://vignette/image.jpg/top-crop/250/140');
+});
+
+test('computes image style properly', function () {
+	var componentMock = this.subject();
+
+	componentMock.updateImageSize(400);
+
+	equal(componentMock.get('imageStyle'), 'height: 219px;');
```
Comment (vforge) [inline test/specs/front/scripts/main/components/TrendingVideosItemComponent.js]: Please add all magic numbers from UnitTests to some variables - so people reading UnitTests alone won't have to dig though code to find why those numbers.

File: test/specs/front/scripts/main/components/TrendingArticlesItemComponent.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+	equal(componentMock.get('currentlyRenderedImageUrl'), 'http://vignette/image.jpg/top-crop/250/140');
+});
+
+test('computes style properly', function () {
+	var componentMock = this.subject();
+
+	componentMock.updateImageSize(400);
+
+	equal(componentMock.get('style'), 'width: 190px;');
```
Comment (vforge) [inline test/specs/front/scripts/main/components/TrendingArticlesItemComponent.js]: Please add all magic numbers from UnitTests to some variables - so people reading UnitTests alone won't have to dig though code to find why those numbers. How is 190px and 106px connected to 400?

File: test/specs/front/scripts/main/components/TrendingArticlesItemComponent.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		imageUrl: 'http://vignette/image.jpg'
+	});
+
+	componentMock.thumbnailer.getThumbURL = function (url, options) {
+		return url + '/' + options.mode + '/' + options.width + '/' + options.height;
+	};
+
+	componentMock.lazyLoadImage();
+
+	equal(componentMock.get('currentlyRenderedImageUrl'), 'http://vignette/image.jpg/top-crop/250/140');
```
Comment (vforge) [inline test/specs/front/scripts/main/components/TrendingArticlesItemComponent.js]: Please add all magic numbers from UnitTests to some variables - so people reading UnitTests alone won't have to dig though code to find why those numbers. How is 140 connected to 250?

File: test/specs/front/scripts/main/components/FeaturedContentItemComponent.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+moduleForComponent('featured-content-item', 'FeaturedContentItemComponent');
+
+test('computes container height properly', function () {
+	var componentMock = this.subject();
+
+	componentMock.updateContainerHeight(400);
+
+	equal(componentMock.get('style'), 'height: 225px;');
```
Comment (vforge) [inline test/specs/front/scripts/main/components/FeaturedContentItemComponent.js]: Please add all magic numbers from UnitTests to some variables - so people reading UnitTests alone won't have to dig though code to find why those numbers. How is 225px connected to 400?

File: test/specs/front/scripts/main/components/CuratedContentItemComponent.js [hunk 5]
```diff
@@ -1,40 +1,54 @@
...
+
+	equal(componentMock.get('thumbUrl'), 'http://vignette/image.jpg/top-crop/400/400');
+});
+
+test('computes image style properly', function () {
+	var componentMock = this.subject();
+
+	componentMock.updateImageSize(400);
+
+	equal(componentMock.get('style'), 'height: 190px; width: 190px;');
```
Comment (vforge) [inline test/specs/front/scripts/main/components/CuratedContentItemComponent.js]: Please add all magic numbers from UnitTests to some variables - so people reading UnitTests alone won't have to dig though code to find why those numbers. How is 190px connected to 400?

File: test/specs/front/scripts/main/components/TrendingVideosItemComponent.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+	componentMock.setProperties({
+		imageWidth: 250,
+		mode: 'top-crop',
+		video: {
+			url: 'http://vignette/image.jpg'
+		}
+	});
+
+	equal(componentMock.get('thumbUrl'), 'http://vignette/image.jpg/top-crop/250/140');
```
Comment (vforge) [inline test/specs/front/scripts/main/components/TrendingVideosItemComponent.js]: Please add all magic numbers from UnitTests to some variables - so people reading UnitTests alone won't have to dig though code to find why those numbers. How is 140 connected to 400?

File: test/specs/front/scripts/main/routes/MainPageCategoryRoute.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+		originalM = M;
+	},
+	teardown: function () {
+		M = originalM;
+	}
+});
+
+test('sets controller properties', function () {
+	// There is so much going on, better to be sure
+	expect(8);
```
Comment (vforge) [inline test/specs/front/scripts/main/routes/MainPageCategoryRoute.js]: Why not 9? :P On serious note, what's with that magic number?

File: test/specs/front/scripts/main/routes/MainPageRoute.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+		equal(dynamicSegment, '~%2560!%2540%2523%2524%2525%255E%2526*()%2520%252B-%253D%257B%257D%255B%255D%257C%253B%253A\'%2522%253C%253E%253F%252C.%252F');
+	};
+
+	routeMock.send('openCuratedContentItem', {
+		type: 'section',
+		label: '~`!@#$%^&*() +-={}[]\|;:\'"<>?,./'
+	});
+});
+
+test('opens curated content category', function () {
```
Comment (vforge) [inline test/specs/front/scripts/main/routes/MainPageRoute.js]: Shouldn't this test's name be connected to checking for encoding `categoryName`?

File: test/specs/front/scripts/main/routes/MainPageRoute.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+moduleFor('route:mainPage', 'MainPageRoute');
+
+test('opens curated content section', function () {
```
Comment (vforge) [inline test/specs/front/scripts/main/routes/MainPageRoute.js]: Shouldn't this test's name be connected to checking for encoding `sectionName`?

File: test/specs/front/scripts/main/routes/MainPageCategoryRoute.js (hunk 7 above)
Comment (rogatty) [inline test/specs/front/scripts/main/routes/MainPageCategoryRoute.js]: It's not a magic number, it's the number of assertions in the test, see http://api.qunitjs.com/expect/

It should be used when there are async operations so I'll remove it anyway.

**Comment by tomasznapieralski** [comment]
LGTM

**Comment by vforge** [comment]
LGTM

**Pull request merged**